### ISSUE TITLE
doesn't work with aot

### ISSUE THREAD

**Issue by ginsleepy** [SuperiorJT/angular2-qrcode#12]
```
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/dir/app'
```

**Comment by ginsleepy** [comment]
```
ERROR in Unexpected value 'QRCodeModule
```

**Comment by janein** [comment]
Hi, I have a similar (the same?) problem.
I'm working on an Ionic app which uses aot-compiling in production-mode.
When I use the module in development mode (no aot) it works fine. But when the aot is enabled (I need this for performance reasons) I get this error message:

`Error: Unexpected value 'QRCodeModule' imported by the module 'AppModule'`

**Comment by SuperiorJT** [comment]
Can you guys try again, but go into your node modules and change the type definitions to have what is in [this PR](https://github.com/SuperiorJT/angular2-qrcode/pull/9/files)? If that is the issue, I can push a fix out for that right away.

**Comment by janein** [comment]
@SuperiorJT I'm sorry but I've tried and this alone doesn't solve the issue. But I guess it may not be bad to include this PR anyway?

**Comment by Brown166** [comment]
I'm using Angular2 CLI and I always get the error "ERROR in QRCodeModule is not an NgModule webpack: Failed to compile" while compiling. So I tried to change the type definitions like SuperiorJT suggested. It solved my problem. I get no error message anymore.

**Comment by Gustav0ar** [comment]
@SuperiorJT that also didn't solve this issue for me when doing that change and compiling with AoT. It just changes the error, now I get:
Can't bind to 'value' since it isn't a known property of 'qr-code'.
1. If 'qr-code' is an Angular component and it has 'value' input, then verify that it is part of this module.

**Comment by janein** [comment]
@Brown166 Did you do any other changes beside this one? Because I can't see any reason why this fix works for you but not for me and @Dillinjer  ... :thinking:

**Comment by Brown166** [comment]
Hi janein,
no I didn't do any other changes. I just modified the file "angular2-qrcode.d.ts" as described above. I tested it several times: So after I delete the folder "node_modules" and I install the modules with npm install again then I get the error when I try to compile the program . If I exchange the "angular2-qrcode.d.ts" with my modified version and I compile the program again everything works well. No errors.

At the moment I'm just using version 1.0.5 not the new version 2.0.0 of angular2-qrcode.

**Comment by delcerrocristian** [comment]
Hi all, i have the same problem:

```
ERROR in Unexpected value 'QRCodeModule in /home/rof/src/.../node_modules/angular2-qrcode/angular2-qrcode.d.ts' imported by the module 'AppModule in /home/rof/src../app.module.ts'

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './../aot/src/app/app.module.ngfactory' in '/home/rof/.../src'
```

**Comment by appzone** [comment]
hi 
i have same problem

ERROR in Template parse errors:
Can't bind to 'value' since it isn't a known property of 'qr-code'.
1. If 'qr-code' is an Angular component and it has 'value' input, then verify t
at it is part of this module.


ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in
 'D:\test\test_ts\Todos2\src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts

**Comment by janein** [comment]
@SuperiorJT I just continued the development on my project and tried to solve this issue again. Now i get the same behaviours as @Dillinjer and @appzone . When I apply the suggested changes to the `angular2-qrcode.d.ts` I get  the same error saying that "value" (or data - tried both) isn't a valid property.
But this seems to be a "better" error than before, so it would be nice if you could apply the changes from the PR #9 .

**Comment by SuperiorJT** [comment]
Marking this as solved with the 2.0.1 release.
